wordpress-mobile/WordPress-Android
# Reblog feature: small customization of site picker

This PR tries to further enhance the #11555 by adding a little step in site picker selection before to continue the reblog flow when a user has more than one site.

| NOTE | IMAGE |
|---|---|
| Adding `CONTINUE` menu disabled at beginning | ![image](https://user-images.githubusercontent.com/47797566/79810282-fcdbe000-8372-11ea-950b-f28e9cc6cdc4.png) |
| Enabling `CONTINUE` menu after first site selection  | ![image](https://user-images.githubusercontent.com/47797566/79810457-6360fe00-8373-11ea-9af9-a02217be746e.png) |
| Search will allow site filtering and on filtered site selection user will be again in a  screen like the first one above  | ![image](https://user-images.githubusercontent.com/47797566/79810850-5abcf780-8374-11ea-8783-f59d591f747c.png)  |

cc @osullivanchris and @iamthomasbishop 

## To test
Be sure to be logged in with a WP.com account having more than one site (you should have enough sites to need to scroll in the site picker so to manual test the scroll to selected scenario)

### Feature Test
- From a reader post select the reblog button
- Check you get access to the site picker like below without any selected site

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/80132154-8ec63180-859b-11ea-9c7f-338fb36417d3.png>
</p>

- Check single site is selected if you tap on it, and you get an action bar like below

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/80132107-7fdf7f00-859b-11ea-9de5-aab7ce257b8e.png>
</p>

- Check you can select other sites and still remain in the same screen as above with an updated action bar title set to the site title
- Check you can go back and come to the initial screen with no site selected
- Long press on a site and check nothing happens (hiding sites capability should not be available)
- Activate search and type some search string to get a filtered list of sites
- Select a site that is at the lower end of the main list
- Check the search view is dismissed and the sites list is scrolled to the selected site
- Tap on the Checkmark icon and check you can complete the reblog flow.

### Smoke Test previous available scenarios
Site Picker is used in `My Site > Switch Site`, `Share to WP`, `Login Epilogue`; Smock test these scenarios to check for eventual regressions. In particular check the Site Hide feature in `Switch Site` picker works as usual.


PR submission checklist:
